overlord/snapstate/check_snap.go

```diff
@@ -62,6 +62,10 @@ func checkSnap(st *state.State, snapFilePath string, curInfo *snap.Info, flags F
 		return err
 	}
 
+	if s.NeedsDevMode() && !flags.DevModeAllowed() {
+		return fmt.Errorf("snap %q has devmode confinement but user has not requested nor overridden", s.Name())
```

Comment:
does this come from a UX doc? it sounds super strange to refer to the "user" this way to me